src/auxDataHashBuilder.c

```diff
@@ -0,0 +1,250 @@
+#include "common.h"
+#include "auxDataHashBuilder.h"
+#include "hash.h"
+#include "cbor.h"
+#include "cardano.h"
+#include "crc32.h"
+#include "bufView.h"
+
+// this tracing is rarely needed
+// so we want to keep it turned off to avoid polluting the trace log
+
+//#define TRACE_AUX_DATA_HASH_BUILDER
+
+#ifdef TRACE_AUX_DATA_HASH_BUILDER
+#define _TRACE(...) TRACE(__VA_ARGS__)
+#else
+#define _TRACE(...)
+#endif // DEVEL
+
+
+// Syntactic sugar
+#define BUILDER_APPEND_CBOR(type, value) \
+	blake2b_256_append_cbor(&builder->auxDataHash, type, value, true); \
+	if (auxDataHashBuilder_isInCatalystRegistrationPayload(builder)) { \
+		blake2b_256_append_cbor(&builder->catalystRegistrationData.payloadHash, type, value, false); \
+	}
+
+#define BUILDER_APPEND_DATA(buffer, bufferSize) \
+	blake2b_256_append_and_trace(&builder->auxDataHash, buffer, bufferSize); \
+	if (auxDataHashBuilder_isInCatalystRegistrationPayload(builder)) { \
+		blake2b_256_append(&builder->catalystRegistrationData.payloadHash, buffer, bufferSize); \
+	}
```

Comment:
I find this quite hard to follow. There are some non-sugared calls to blake2b below, too (at least one). What about having macros like `APPEND_AUXDATA_CBOR`, `APPEND_AUXDATA_BUFFER`, `APPEND_CATALYST_CBOR`, `APPEND_BOTH_CBOR` (defined as composition of the two) etc.? Then at least one can use editor search function to see what is added to the respective hashes.

Reply:
This could work but I feel like it would be kind of cumbersome to manage/extend as soon as we'd want to add anything else, ultimately creating a tech debt for whoever would add support for some other kind of metadata which is plausible to happen in the foreseeable future.

I ended up creating a binary mask backed by an enum, specifying what hash builders to select which allows to explicitly select the relevant hash builders by their codename when using the macro - I just pushed it. What do you think?